insights/core/spec_factory.py

```diff
@@ -290,8 +289,19 @@ def load(self):
             self.rc = rc
             return out
 
+        fsize = os.stat(self.path).st_size
         with safe_open(self.path, "r", encoding=encoding, errors="surrogateescape") as f:
-            return [l.rstrip("\n") for l in f]
+            if fsize > MAX_CONTENT_SIZE:
+                # read the last ``MAX_CONTENT_SIZE`` MB only
+                f.seek(fsize - MAX_CONTENT_SIZE)
+                log.debug("Extra-huge file is truncated %s", self.relative_path)
+                content = [l.rstrip("\n") for l in f][1:]  # discard the first line which is broken
+            else:
+                content = [l.rstrip("\n") for l in f]
+            if not isinstance(self.ctx, HostContext) and self._filters:
+                # Post-filtering ONLY when processing data
+                return AllowFilter.filter_content(content[::-1], self._filters)[::-1]
```

Comment:
For large `content`, the `[::-1]` list reverse action would construct a new list object and cost quite some memory. Is it possible to pass the content as itself without reverse to `AllowFilter.filter_content`, and do a bottom-up filter check in it.